code/modules/power/lighting.dm

```diff
@@ -642,14 +642,18 @@
 		if(istype(H))
 			var/datum/species/ethereal/eth_species = H.dna?.species
 			if(istype(eth_species))
+				var/datum/species/ethereal/E = H.dna.species
+				if(E.drain_time > world.time)
+					return
 				to_chat(H, "<span class='notice'>You start channeling some power through the [fitting] into your body.</span>")
-				while(do_after(user, 20, target = src))
-					if(eth_species.ethereal_charge >= ETHEREAL_CHARGE_FULL)
-						to_chat(H, "<span class='notice'>You are now fully charged.</span>")
-						break
-					else
+				E.drain_time = world.time + 30
```

Comment:
Will the charging here not loop anymore?

Reply:
nope, no more spam draining